### PR TITLE
Implement basic TUI game

### PR DESCRIPTION
**Pull request by Fanarito**
Simple implementation that uses the generalized UI to play tic tac toe

### CODE REVIEW

File: src/main/hopur0/games/tictactoe/Game.java [hunk 1]
```diff
@@ -1,7 +1,55 @@
...
+    board = new Board();
+    initializePlayers();
+    gameLoop();
+  }
+
+  private void initializePlayers() {
+    String firstPlayerName = ui.getString("Enter first player name:");
+    String secondPlayerName = ui.getString("Enter second player name:");
+    players = new Player[] {
+      new Player(firstPlayerName, 'x'),
```
Comment (dadisigursveinn) [inline src/main/hopur0/games/tictactoe/Game.java]: maybe request what symbol users want to use?

File: src/main/hopur0/games/tictactoe/Game.java [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+      } while (!board.validTile(selectedTile) || board.getTile(selectedTile) != null);
+
+      board.setTile(players[currentPlayer], selectedTile);
+      currentPlayer = (currentPlayer + 1) % 2;
     }
+
+    ui.showMessage("----------------------------");
+    ui.showMessage("Final state:" + System.lineSeparator() + board.toString());
+
+    if (board.full()) {
```
Comment (dadisigursveinn) [inline src/main/hopur0/games/tictactoe/Game.java]: If a player wins and fills the board at the same time would't that cause the code to display a draw?

**Review by dadisigursveinn — CHANGES_REQUESTED** [review]
indentation in Game.java not consistent with other classes.
There are no tests for the Game class

File: src/main/hopur0/games/tictactoe/Game.java (hunk 1 above)
Comment (Fanarito) [reply src/main/hopur0/games/tictactoe/Game.java]: In our design report, we define that X and O are commonly used, probably an unnecessary complication

**Review by hrafnlogi — APPROVED** [review]

**Review by dadisigursveinn — APPROVED** [review]
lack of unit tests #14 opened in a new issue

**Pull request merged**